### PR TITLE
RAB-1087: Add Google Cloud Storage support (front)

### PR DESCRIPTION
**Pull request by axelhms**
**Description (for Contributor and Core Developer)**

In this PR, I've added support for Google Cloud remote server.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/GoogleCloudStorageConfigurator.test.tsx [hunk 1]
```diff
@@ -0,0 +1,345 @@
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import {screen, act} from '@testing-library/react';
+import {renderWithProviders, ValidationError} from '@akeneo-pim-community/shared';
+import {GoogleCloudStorage, LocalStorage} from '../model';
+import {GoogleCloudStorageConfigurator} from './GoogleCloudStorageConfigurator';
+
+beforeEach(() => {
```
Comment (ghoussard) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/GoogleCloudStorageConfigurator.test.tsx:8]: We should add a card in our backlog (RAC or RAB?) to mock hooks, not the fetch
(Don't worry about that in this PR)

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/GoogleCloudStorageConfigurator.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      />
+      <TextField
+        required={true}
+        value={storage.bucket}
+        label={translate('pim_import_export.form.job_instance.storage_form.bucket.label')}
+        placeholder={translate('pim_import_export.form.job_instance.storage_form.bucket.placeholder')}
+        onChange={(bucket: string) => onStorageChange({...storage, bucket})}
+        errors={filterErrors(validationErrors, '[bucket]')}
+      />
+      <CheckStorageForm>
```
Comment (ghoussard) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/GoogleCloudStorageConfigurator.tsx:71]: We should add a card in our backlog (RAC or RAB?) to extract it in a component
(Don't worry about that in this PR)

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts [hunk 3]
```diff
@@ -35,16 +35,24 @@ type MicrosoftAzureStorage = {
...
 };
 
-type Storage = LocalStorage | SftpStorage | AmazonS3Storage | MicrosoftAzureStorage | NoneStorage;
+type Storage = LocalStorage | SftpStorage | AmazonS3Storage | MicrosoftAzureStorage | GoogleCloudStorage | NoneStorage;
 
-type StorageType = 'none' | 'local' | 'sftp' | 'amazon_s3' | 'microsoft_azure';
+type StorageType = 'none' | 'local' | 'sftp' | 'amazon_s3' | 'microsoft_azure' | 'google_cloud';
 
-const STORAGE_TYPES = ['none', 'sftp', 'amazon_s3', 'microsoft_azure'];
+const STORAGE_TYPES = ['none', 'sftp', 'amazon_s3', 'microsoft_azure', 'google_cloud'];
```
Comment (ghoussard) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts:55]: I think there is a reason but it appears weird that `STORAGE_TYPES` does not contain `local` 🤔
(Again, we should handle that in another card)

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/hooks/useCheckStorageConnection.ts [hunk 4]
```diff
@@ -27,7 +32,15 @@ const isMicrosoftAzureConnectionFieldFulfilled = (storage: MicrosoftAzureStorage
   return '' !== storage.connection_string && '' !== storage.container_name && '' !== storage.file_path;
 };
 
-const useCheckStorageConnection = (storage: SftpStorage | AmazonS3Storage | MicrosoftAzureStorage) => {
+const isGoogleCloudConnectionFieldFulfilled = (storage: GoogleCloudStorage): boolean => {
```
Comment (ghoussard) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/hooks/useCheckStorageConnection.ts:35]: Would be nice if you add tests to test directly those functions (`is[Storage]ConnectionFieldFulfilled`) 👍🏻 
(Can be in another card if it bother you)

**Review by ghoussard — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/GoogleCloudStorageConfigurator.tsx (hunk 2 above)
Comment (ValentinMumble) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/GoogleCloudStorageConfigurator.tsx:71]: Agreed, I believe Axel already created the card https://akeneo.atlassian.net/browse/RAB-1236

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/model.test.ts [hunk 5]
```diff
@@ -52,6 +55,14 @@ const microsoftAzureStorage: MicrosoftAzureStorage = {
   file_path: '/tmp/test.xlsx',
 };
 
+const googleCloudStorage: GoogleCloudStorage = {
+  type: 'google_cloud',
+  file_path: '/tmp/test.xlsx',
+  project_id: 'eu-west-3',
+  service_account: 'a_service_account',
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/model.test.ts:62]: Maybe we can also use a fake JSON string here too, to better reflect a real use-case

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts (hunk 3 above)
Comment (ValentinMumble) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts:55]: Should be handled with the local storage card IMO (https://akeneo.atlassian.net/browse/RAB-665)

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/GoogleCloudStorageConfigurator.test.tsx (hunk 1 above)
Comment (axelhms) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/GoogleCloudStorageConfigurator.test.tsx:8]: Good idea, added the card in our backlog https://akeneo.atlassian.net/jira/software/projects/RAB/boards/282/backlog?selectedIssue=RAB-1242

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/hooks/useCheckStorageConnection.ts (hunk 4 above)
Comment (axelhms) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/hooks/useCheckStorageConnection.ts:35]: Same for this one [RAB-1242](https://akeneo.atlassian.net/jira/software/projects/RAB/boards/282/backlog?selectedIssue=RAB-1242)

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts [hunk 6]
```diff
@@ -35,16 +35,24 @@ type MicrosoftAzureStorage = {
   container_name: string;
 };
 
+type GoogleCloudStorage = {
+  type: 'google_cloud';
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts:39]:
```suggestion
  type: 'google_cloud_storage';
```

Comment (axelhms) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts:39]: 🦅

**Pull request merged**